### PR TITLE
Add a `node` label to agent metrics.

### PR DESCRIPTION
**Pull request by diversario**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Exposing node name allows correlating Cilium metrics to node metrics for
more advanced recording rules/alerts, for example: ignore endpoint
health on nodes aged less than 2 minutes.

### CODE REVIEW

File: install/kubernetes/cilium/charts/agent/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -41,5 +41,10 @@ spec:
     interval: 10s
     honorLabels: true
     path: /metrics
+    relabelings:
+    - replacement: ${1}
```
Comment (sayboras) [inline install/kubernetes/cilium/charts/agent/templates/servicemonitor.yaml:45]: nit: this is default value, which can be removed. Feel free to ignore.

**Review by sayboras — COMMENTED** [review]
Maybe a question to you and others :)  

Do we need to add for other ServiceMonitor as well or will it be case by case? The best case that I have in mind is to expose `relabel configs` in values.yaml, so that users can just adjust values based on their needs, I don't know how to achieve it in concise way now though.

[1] https://prometheus.io/docs/prometheus/latest/configuration/configuration/

File: install/kubernetes/cilium/charts/agent/templates/servicemonitor.yaml (hunk 1 above)
Comment (joestringer) [reply install/kubernetes/cilium/charts/agent/templates/servicemonitor.yaml:45]: @sayboras do you want @diversario to remove this line or not? If we think it should be here, then I think we can resolve this point. If this line should not be there, please use the "request changes" approach to make it clear that the contributor will need to address this prior to accepting this patch.

**Review by joestringer — CHANGES_REQUESTED** [review]
Please either write a release note in the section of the description with the following text, or remove that section:

```
<!-- Enter the release note text here if needed or remove this section! -->
```

This will ensure that the release notes will mention your PR submission with a clear description.

Other than this, if it's passing CI and helps solve what you're trying to do then I think this makes sense to merge :+1:

File: install/kubernetes/cilium/charts/agent/templates/servicemonitor.yaml (hunk 1 above)
Comment (diversario) [reply install/kubernetes/cilium/charts/agent/templates/servicemonitor.yaml:45]: Sorry about lack of response!

Re this line: I understand it's redundant but IMO it's ok to keep as it for people who're reading this and trying to understand how things work (definitely helped me).

**Review by sayboras — APPROVED** [review]
LGTM :100:

**Review by joestringer — APPROVED** [review]

**Pull request merged**